ckanext/privatedatasets/tests/test_actions.py

```diff
@@ -163,13 +161,9 @@ def _get_action(action):
     ])
     def test_add_users(self, users_info, datasets_not_found, not_updatable_datasets, allowed_users=[]):
 
-        parse_result = {'users_datasets': []}
+        parse_result = {'users_datasets': [{'user': user, 'datasets': users_info[user]} for user in users_info]}
```

Comment:
To be honest, i did it that way thanks to Fran, i have only seen that kind of list comprehesion in haskell and didnt know that python also support it.